smaug/python/ops/control_flow_ops.py

```diff
@@ -30,9 +30,77 @@ def merge(input_tensors, name="merge"):
   Returns:
     A tensor that the available input tensor forwards to.
   """
-  return add_node(
-      name=name,
-      op=Merge,
-      input_tensors=input_tensors,
+  return common.add_node(
+      name=name, op=types_pb2.Merge, input_tensors=input_tensors,
       output_tensors_dims=[input_tensors[0].shape.dims],
       output_tensor_layout=input_tensors[0].shape.layout)[0]
+
+def cond(predication, true_fn, false_fn, name="cond"):
+  """A conditional operator.
+
+  This operator provides the capability of doing if-else statement. Depending on
+  the predication value, either the True or the False body of the operator will
+  be executed.
+
+  Args:
+    predication: A predication tensor of value 0 or 1, determining which path to
+      execute.
+    true_fn: The callable to be performed if `predication` is 1.
+    false_fn: The callable to be performed if `predication` is 0.
+
+  Returns:
+    The tensors returned by either true_fn or false_fn.
+  """
+
+  def _insert_switch_nodes(predication, branch, graph):
+    """Insert switch nodes for external tensros in the subgraph."""
+    nodes = [node for node in graph.get_nodes() if node.op != types_pb2.Data]
+    internal_tensors = []
+    for node in nodes:
+      internal_tensors.extend([tensor.name for tensor in node.output_tensors])
+    for node in nodes:
+      for i, tensor_proto in enumerate(node.input_tensors):
+        # If any input tensor of the graph appear in the graph workspace, then
+        # this tensor is an external to the graph and we create a switch node
+        # for it.
+        # Don't create switch node for an existing one.
+        if node.op == types_pb2.Switch:
+          continue
+        if tensor_proto.name not in internal_tensors:
+          source_node = graph.get_node(node.parents[i], True)
+          tensor = tensor_utils.from_tensor_proto(tensor_proto)
+          if source_node is not None:
+            tensor.source = (source_node, node.src_tensors_indices[i])
+          switch_result = switch(tensor, predication)[branch]
+          switch_result.to_tensor_proto(node.input_tensors[i])
+          # Update the node with the switch node as its new parent.
```

Comment:
This comment would make sense if placed one line earlier.

Reply:
Done.